disco/gateway/client.py

```diff
@@ -266,14 +267,21 @@ def run(self):
         gevent.spawn(self.connect_and_run)
         self.ws_event.wait()
 
-    def request_guild_members(self, guild_id_or_ids, query=None, limit=0):
+    def request_guild_members(self, guild_id_or_ids, query=None, limit=0, user_id_or_ids=None, presences=None):
         """
         Request a batch of Guild members from Discord. Generally this function
         can be called when initially loading Guilds to fill the local member state.
+        When calling this function, if user_id_or_ids is passed then query will be ignored.
         """
-        self.send(OPCode.REQUEST_GUILD_MEMBERS, {
+        payload = {
             # This is simply unfortunate naming on the part of Discord...
             'guild_id': guild_id_or_ids,
-            'query': query or '',
             'limit': limit,
-        })
+        }
+        payload.update(optional(
+            # This is simply Discord sticking to an unfortunate naming scheme...
+            user_ids=user_id_or_ids,
+            query=query or '' if user_id_or_ids is None else None,
```

Comment:
Whats the behavior on Discord's side if both a `query` and `user_ids` is sent? Ideally this function should raise an exception if the caller passes invalid data (e.g. both query and user_id_or_ids), or we should consider splitting this up into two functions.

Reply:
As far as behaviour goes, Discord ignores user_ids when it is sent with query.

Reply:
In that case lets either split this up into a new function, or raise an exception if both query and user_ids is passed.